server/events/vcs/gitlab_client.go

```diff
@@ -49,25 +49,28 @@ func NewGitlabClient(hostname string, token string, logger *logging.SimpleLogger
 
 	// If not using gitlab.com we need to set the URL to the API.
 	if hostname != "gitlab.com" {
-		u, err := url.Parse(hostname)
+		//We assume the url will be over HTTPS if the user doesn't specify a scheme
+		absoluteURL := strings.TrimSuffix(hostname, "/")
```

Comment:
I prefer doing the suffix trim right before we construct `apiURL`. Then it's clear we're doing it to aid URL construction vs. up here I'm wondering why you're trimming it.